libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PathLookupImpl.java

```diff
@@ -0,0 +1,147 @@
+            return pidFile == null ? null : Stream.of(pidFile);
+        } else {
+            throw new IllegalStateException("unknown base dir [" + baseDir + "]");
+        }
+    }
+
+    @Override
+    public Stream<Path> resolveRelativePaths(BaseDir baseDir, Path relativePath) {
+        if (baseDir == BaseDir.HOME) {
+            return Stream.of(homeDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.DATA) {
+            return Arrays.stream(dataDirs).map(dataDirPath -> dataDirPath.resolve(relativePath));
+        } else if (baseDir == BaseDir.SHARED_REPO) {
+            return Arrays.stream(sharedRepoDirs).map(sharedRepoDir -> sharedRepoDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.CONFIG) {
+            return Stream.of(configDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.LIB) {
+            return Stream.of(libDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.MODULES) {
+            return Stream.of(modulesDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.PLUGINS) {
+            return Stream.of(pluginsDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.LOGS) {
+            return Stream.of(logsDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.TEMP) {
+            return Stream.of(tempDir.resolve(relativePath));
+        } else if (baseDir == BaseDir.PID) {
+            return pidFile == null ? null : Stream.of(pidFile.resolve(relativePath));
+        } else {
+            throw new IllegalStateException("unknown base dir [" + baseDir + "]");
+        }
+    }
+
+    @Override
+    public Stream<Path> resolveSettingPaths(BaseDir baseDir, String settingName) {
+        Stream<Path> relativePaths = settingResolver.apply(settingName)
+            .filter(s -> s.toLowerCase(Locale.ROOT).startsWith("https://") == false)
+            .distinct()
+            .map(Path::of);
+        if (baseDir == BaseDir.HOME) {
+            return relativePaths.map(homeDir::resolve);
+        } else if (baseDir == BaseDir.DATA) {
+            return relativePathsCombination(dataDirs, relativePaths);
+        } else if (baseDir == BaseDir.SHARED_REPO) {
+            return relativePathsCombination(sharedRepoDirs, relativePaths);
+        } else if (baseDir == BaseDir.CONFIG) {
+            return relativePaths.map(configDir::resolve);
+        } else if (baseDir == BaseDir.LIB) {
+            return relativePaths.map(libDir::resolve);
+        } else if (baseDir == BaseDir.MODULES) {
+            return relativePaths.map(modulesDir::resolve);
+        } else if (baseDir == BaseDir.PLUGINS) {
+            return relativePaths.map(pluginsDir::resolve);
+        } else if (baseDir == BaseDir.LOGS) {
+            return relativePaths.map(logsDir::resolve);
+        } else if (baseDir == BaseDir.TEMP) {
+            return relativePaths.map(tempDir::resolve);
+        } else if (baseDir == BaseDir.PID) {
```

Comment:
PID seems like a special case. It's not a dir, so maybe it doesn't belong in the enum. Rather, it could be a special method just for pid on PathLookup, or have a separate enum for special files? And in that case we don't want any resolution, it's more just lookup up the (potentially many? in the test case) file paths for that constant.

Reply:
Removed.